docs/architecture.md

```diff
@@ -0,0 +1,211 @@
+          import { useSignInRequest } from '@/modules/auth/hooks/use-sign-in-request';
+          import type { BeautifulComponentProps } from './beautiful-component.interface';
+         
+          import { Root } from './beautiful-component.styled';
+         
+          export const BeautifulComponent: FC<BeautifulComponentProps> = () => {}
+         ```
+   - Use `type` prefix for all type imports
+     - Correct:
+        ```ts
+        import { type FC } from 'react';
+        import type { BeautifulComponentProps } from './beautiful-component.interface'
+         ```
+     - Incorrect:
+        ```ts
+        import { FC } from 'react';
+        import { BeautifulComponentProps } from './beautiful-component.interface';
+       ```
+> Note! If one of rules above must be violated for an important reason, please leave `todo` comment in the code and create an issue in the GitHub
+
+## <a name="folder-structure"></a> Folder structure
+
+1. **assets**
+
+    Static files, such as icons, images, fonts, etc... Each kind of asset should have an individual folder.
+    ```
+    src/
+      assets/
+        icons/
+          my-icon.svg
+        images/
+          my-image.jpg
+        fonts/
+   ```
+2. **declarations**
+
+    Global type declarations. 
+
+3. **modules**
+
+    Module - independent project entity that contains ui, data, domain logic of this entity. 
+    ```
+    src/
+      modules/
+        user/
+        location/
+        auth/
+   ```
+4. **modules/module-name/ui**
+   
+   Ui folder contains visual components and containers.
+
+    ```
+   src/
+     modules/
+       user/
+         ui/
+           components/
+             user-avatar/*
+           containers/
+             user-profile-edit-form/*
+         data/
+         domain/
+   ```
+   
+5. **modules/module-name/data**
+
+   Module can contain:
+    - Hooks
+    - Component
+    - Containers
+    - Constants
+    - Data types
+    - Global storage (redux) configuration - actions, reducers, selectors
+    - Mappers
+
+    ```
+    src/
+      modules/
+        user/
+          components/
+          containers/
+          hooks/
+          constants/
+          types/
+          global-storage/
+          services/
+   ```
+   
+6. **styles**
+
+     Theme, global styles, style helpers, etc...
+
+## <a name="service"></a> Core libraries
+
+1. todo
+
+## <a name="component"></a> Component
+
+Component it's a simple, reusable ui element that display view depends **only** on props. We create folder with for each component. The component does not have access to global store and domain logic.
+
+Each component consists of next folders:
+1. **component-name.interface.ts** - includes types that used in this component (prop types etc...)
+2. **component-name.styled.ts** - includes styled elements
+3. **component-name.tsx** - component markup (jsx)
+4. **component-name.stories.tsx - storybook story
```

Comment:
You forgot to close **. Also I think we need to add here `component-name.test.tsx`